tools/cmd/api-generator/convert_commands.go

```diff
@@ -101,37 +101,67 @@ func extractSunsetDate(extensions map[string]any) *time.Time {
 type operationExtensions struct {
 	skip             bool
 	operationID      string
+	shortOperationID string
 	operationAliases []string
 }
 
 func extractExtensionsFromOperation(operation *openapi3.Operation) operationExtensions {
 	ext := operationExtensions{
 		skip:             false,
 		operationID:      operation.OperationID,
+		shortOperationID: "",
 		operationAliases: []string{},
 	}
 
-	if extensions, okExtensions := operation.Extensions["x-xgen-atlascli"].(map[string]any); okExtensions && extensions != nil {
-		if extSkip, okSkip := extensions["skip"].(bool); okSkip && extSkip {
-			ext.skip = extSkip
-		}
+	processAtlasCLIExtensions(&ext, operation.Extensions)
+	processShortOperationIDOverride(&ext, operation)
 
-		if extAliases, okExtAliases := extensions["command-aliases"].([]any); okExtAliases && extAliases != nil {
-			for _, alias := range extAliases {
-				if sAlias, ok := alias.(string); ok && sAlias != "" {
-					ext.operationAliases = append(ext.operationAliases, sAlias)
-				}
+	return ext
+}
+
+func processAtlasCLIExtensions(ext *operationExtensions, extensions map[string]any) {
+	atlasCLIExt, ok := extensions["x-xgen-atlascli"].(map[string]any)
+	if !ok || atlasCLIExt == nil {
+		return
+	}
+
+	if extSkip, okSkip := atlasCLIExt["skip"].(bool); okSkip && extSkip {
+		ext.skip = extSkip
+	}
+
+	if extAliases, okExtAliases := atlasCLIExt["command-aliases"].([]any); okExtAliases && extAliases != nil {
+		for _, alias := range extAliases {
+			if sAlias, ok := alias.(string); ok && sAlias != "" {
+				ext.operationAliases = append(ext.operationAliases, sAlias)
 			}
 		}
+	}
 
-		if overrides := extractOverrides(operation.Extensions); overrides != nil {
-			if overriddenOperationID, ok := overrides["operationId"].(string); ok && overriddenOperationID != "" {
-				ext.operationID = overriddenOperationID
-			}
+	if overrides := extractOverrides(extensions); overrides != nil {
+		if overriddenOperationID, ok := overrides["operationId"].(string); ok && overriddenOperationID != "" {
+			ext.operationID = overriddenOperationID
 		}
 	}
+}
 
-	return ext
+func processShortOperationIDOverride(ext *operationExtensions, operation *openapi3.Operation) {
+	shortOperationID, ok := operation.Extensions["x-xgen-operation-id-override"].(string)
+	if !ok || shortOperationID == "" {
+		return
+	}
+
+	ext.shortOperationID = shortOperationID
+
+	// Only add original operation ID to aliases if there's no Atlas CLI operation ID override
+	overrides := extractOverrides(operation.Extensions)
+	if overrides == nil {
+		ext.operationAliases = append(ext.operationAliases, strcase.ToLowerCamel(operation.OperationID))
```

Comment:
Why do we add the original OperationID to the extensions?

AFAIK, the goal of these methods is to extract the extensions only and apply the business logic later.
Running business logic here (adding `operation.OperationID` to aliases) should happen outside of the `extractExtensionsFromOperation` method family.